python/lsst/summit/utils/tmaUtils.py

```diff
@@ -150,11 +165,88 @@ def getAzimuthElevationDataForEvent(client, event, prePadding=0, postPadding=0):
                                prePadding=prePadding,
                                postPadding=postPadding)
 
+    azValues = azimuthData['actualPosition'].values
+    elValues = elevationData['actualPosition'].values
+    azDemand = azimuthData['demandPosition'].values
+    elDemand = elevationData['demandPosition'].values
+
+    azError = (azValues - azDemand) * 3600
+    elError = (elValues - elDemand) * 3600
+
+    azimuthData['azError'] = azError
+    elevationData['elError'] = elError
+
     return azimuthData, elevationData
 
 
-def plotEvent(client, event, fig=None, prePadding=0, postPadding=0, commands={},
-              azimuthData=None, elevationData=None):
+def filterBadValues(values, maxDelta=0.1):
+    """Filter out bad values from a dataset, replacing them in-place.
+
+    This function replaces non-physical points in the dataset with an
+    extrapolation of the preceding two values. No more than 3 successive data
+    points are allowed to be replaced. Minimum length of the input is 3 points.
+
+    Parameters
+    ----------
+    values : `list` or `np.ndarray`
+        The dataset containing the values to be filtered.
+    maxDelta : `float`, optional
+        The maximum allowed difference between consecutive values. Values with
+        a difference greater than `maxDelta` will be considered as bad values
+        and replaced with an extrapolation.
+
+    Returns
+    -------
+    nBadPoints : `int`
+        The number of bad values that were replaced out.
+    """
+    # Find non-physical points and replace with extrapolation. No more than 3
```

Comment:
Better make that a kwarg with a default 3 then, noting it's another 20+ lines before the value actually appears.